### ISSUE TITLE
Add AnyTypeOf matcher

### ISSUE THREAD

**Issue by tfrommen** [padraic/mockery#500]
With the current codebase, I can check an individual argument either against a specific type, or against a set of concrete values:

```php
$some_object = Mockery::mock( 'SomeClass' );
$some_object->shouldReceive( 'foo' )->with( Mockery::type( 'int' ) );
$some_object->foo( 1 );

$some_object->shouldReceive( 'foo' )->with( Mockery::anyOf( 0, 1, 2 ) );
$some_object->foo( 1 );
```

What's missing, however, is *checking against a set of types* so one could do something like the following:

```php
$some_object = Mockery::mock( 'SomeClass' );
$some_object->shouldReceive( 'foo' )->with( Mockery::anyTypeOf( 'int', 'string' ) );
$some_object->foo( 1 );
```

I'll send a pull request for this in a minute.

**Comment by aik099** [comment]
It's already possible by using Hamcrest (see https://github.com/hamcrest/hamcrest-php) matchers. Mockery have native support for them.

**Comment by tfrommen** [comment]
Closed, just like #501 was.
